File: golang/odd_number.go
package main

// findOddNumber finds the integer that appears an odd number of times in the slice.
func findOddNumber(arr []int) int {
	count := make(map[int]int)

	// Count the frequency of each number
	for _, num := range arr {
		count[num]++
	}

	// Find the number that appears an odd number of times
	for num, freq := range count {
		if freq%2 != 0 {
			return num
		}
	}

	// Fallback in case no odd frequency number is found (shouldn't happen with valid input)
	panic("No integer appears an odd number of times")
}

File: javascript/odd-number.js
/**
 * Find the integer that appears an odd number of times in the array.
 * @param {number[]} arr - The input array of integers
 * @return {number} - The integer that appears an odd number of times
 */
export default function findOdd(arr) {
  const count = {};

  // Count the frequency of each number
  for (let num of arr) {
    count[num] = (count[num] || 0) + 1;
  }

  // Find the number that appears an odd number of times
  for (let num in count) {
    if (count[num] % 2 !== 0) {
      return Number(num);
    }
  }
}

// module.exports = findOdd;
